PennMobileShared/Dining/Models/DiningVenue+Extensions.swift

```diff
@@ -22,24 +22,34 @@ public extension VenueType {
 public extension DiningVenue {
 
     // MARK: - Venue Status
-    var mealsToday: Day? {
+    var mealsToday: [Meal] {
+        
+        return mealsOnDate(date: Date())
+    }
+    
+    func mealsOnDate(date: Date) -> [Meal] {
         let dateFormatter = DateFormatter()
         dateFormatter.dateFormat = "yyyy-MM-dd"
+        
         return self.days.first(where: { day in
-            day.date == dateFormatter.string(from: Date())
-        })
+            day.date == dateFormatter.string(from: date)
+        })?.meals.sorted(by: { el1, el2 in
+            return el1.starttime > el2.starttime
+        }) ?? []
     }
 
     var isOpen: Bool {
-        guard let mealsToday = mealsToday else { return false }
-        for meal in mealsToday.meals where meal.isCurrentlyServing {
+        if (mealsToday == []) { return false }
```

Comment:
```suggestion
        if mealsToday.isEmpty { return false }
```

And so on going forward